packages/databricks-vscode/src/vscode-objs/StateStorage.ts

```diff
@@ -1,73 +1,100 @@
-            false
-        );
-    }
+    "databricks.wsfs.skipSwitchToWorkspace": withType<boolean>()({
+        location: "workspace",
+        defaultValue: false,
+    }),
 
-    set skipSwitchToWorkspace(value: boolean) {
-        this.context.workspaceState.update(
-            "databricks.wsfs.skipSwitchToWorkspace",
-            value
-        );
-    }
+    "databricks.autocompletion.skipConfigure": withType<boolean>()({
+        location: "workspace",
+        defaultValue: false,
+    }),
 
-    get skipAutocompleteConfigure() {
-        return this.context.workspaceState.get(
-            "databricks.autocompletion.skipConfigure",
-            false
-        );
-    }
+    "databricks.fixedRandom": withType<number>()({
+        location: "global",
+        getter: (storage, value) => {
+            if (value === undefined) {
+                value = Math.random();
+                storage.set("databricks.fixedRandom", value);
+            }
+            return value;
+        },
+    }),
 
-    set skipAutocompleteConfigure(value: boolean) {
-        this.context.workspaceState.update(
-            "databricks.autocompletion.skipConfigure",
-            value
-        );
-    }
+    "databricks.fixedUUID": withType<string>()({
+        location: "workspace",
+        getter: (storage, value) => {
+            if (value === undefined) {
+                value = randomUUID();
+                storage.set("databricks.fixedUUID", value);
+            }
+            return value;
+        },
+    }),
 
-    get skippedEnvsForDbConnect() {
-        return this.context.globalState.get<string[]>(
-            "databricks.debugging.skipDbConnectInstallForEnvs",
-            []
-        );
-    }
+    "databricks.debugging.skipDbConnectInstallForEnvs": withType<string[]>()({
+        location: "global",
+        defaultValue: [],
+        setter: (storage, value) => {
+            if (value === undefined || value.length === 0) {
+                return undefined;
+            }
+            const currentEnvs: string[] = storage.get(
+                "databricks.debugging.skipDbConnectInstallForEnvs"
+            );
+            if (!currentEnvs.includes(value[0])) {
+                currentEnvs.push(value[0]);
+            }
+            return currentEnvs;
+        },
+    }),
 
-    skipDbConnectInstallForEnv(value: string) {
-        const currentEnvs = this.skippedEnvsForDbConnect;
-        if (!currentEnvs.includes(value)) {
-            currentEnvs.push(value);
-        }
-        this.context.globalState.update(
-            "databricks.debugging.skipDbConnectInstallForEnvs",
-            currentEnvs
-        );
-    }
+    "databricks.lastInstalledExtensionVersion": withType<string>()({
+        location: "workspace",
+        defaultValue: "0.0.0",
+    }),
+};
+
+type ValueType<K extends keyof typeof Keys> = (typeof Keys)[K]["_type"];
+type GetterReturnType<D extends KeyInfo<any>> = D extends {getter: infer G}
```

Comment:
Does it make sense for a getter type to be different from ValueType?

Reply:
Getters can return `ValueType | undefined` or just `ValueType`. I wanted to account for both the cases, and have the autocompletion NOT complete return as undefined if the getter always returns a value or if the defaultValue is defined. 

This is why we also require the nested withType function. This allows us to have getters which extend the standard getter type in the KeyInfo and potentially NOT return undefined. If we don't do this, we will have to type data as any (makes defining new keys type unsafe) or we will lose out on this information about getters.